src/libraries/Microsoft.Bcl.TimeProvider/ref/Microsoft.Bcl.TimeProvider.cs

```diff
@@ -8,15 +8,14 @@ namespace System
     public abstract class TimeProvider
     {
         public static TimeProvider System { get; }
-        protected TimeProvider(long timestampFrequency) { throw null; }
-        public abstract System.DateTimeOffset UtcNow { get; }
-        public System.DateTimeOffset LocalNow { get; }
-        public abstract System.TimeZoneInfo LocalTimeZone { get; }
-        public long TimestampFrequency { get; }
-        public static TimeProvider FromLocalTimeZone(System.TimeZoneInfo timeZone) { throw null; }
-        public abstract long GetTimestamp();
+        protected TimeProvider() { throw null; }
+        public virtual System.DateTimeOffset GetUtcNow() { throw null; }
+        public System.DateTimeOffset GetLocalNow() { throw null; }
+        public virtual System.TimeZoneInfo LocalTimeZone { get; }
+        public virtual long TimestampFrequency { get; }
+        public virtual long GetTimestamp() { throw null; }
         public TimeSpan GetElapsedTime(long startingTimestamp, long endingTimestamp) { throw null; }
-        public abstract System.Threading.ITimer CreateTimer(System.Threading.TimerCallback callback, object? state, System.TimeSpan dueTime, System.TimeSpan period);
+        public virtual System.Threading.ITimer CreateTimer(System.Threading.TimerCallback callback, object? state, System.TimeSpan dueTime, System.TimeSpan period) { throw null; }
```

Comment:
Question: Why would you want to throw null here? I'd expect NotImplementedException.

Reply:
This is the reference assembly. It should never be used at runtime, and the goal is to define just the signatures while keeping the size as small as possible. throw null is what we do in the ref assemblies everywhere.

Reply:
Thanks, that makes sense!